MTM-64421 Requirement for max header size (#3711)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/cumulocity-2025-62-0-microservices-sdk-allowing-bigger-headers-in-request.md

---

> date: 2025-10-09

Reviewer: ```suggestion
date: 
```

Author: the date should be empty?

Reviewer: Yes, the date field should always be left empty as it is filled in automatically when the related software has been deployed to eu-latest.

title: Increase of max request header size
product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e

>     label: Announcement

Reviewer: Please change this to Improvement or Feature (with the related value). Announcements are only used for changes that have not yet taken place.

Author: I've changed it to Improvement.

component:
  - value: component-Sv2buFZ5I
    label: Microservice SDK
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-64421
version: 2025.62.0
---

> To allow the usage of larger headers in the requests, in microservices SDK we have set the property server.max-http-request-header-size to 24KB. Previous value was 8KB.

Reviewer 2: question: Is it a default value which can be **changed** by the customer using SDK, or such change would not make sense, because our infrastructure (eg. ontop-lb) will not pass more than 24KB?

Author: In the last version of the implementation (it changed many times), it is the default value that we set, and it can be changed. However, it does not really make sense, as we have this restriction on the ontoplb side , and also in core.

> To allow the usage of larger headers in the requests, in microservices SDK we have set the property server.max-http-request-header-size to 24KB. Previous value was 8KB.

Reviewer: ```suggestion
To allow the usage of larger headers in requests, the property `server.max-http-request-header-size` has been increased to 24KB in the Microservice SDK. Previously, the value was 8KB.
```

> To allow the usage of larger headers in the requests, in microservices SDK we have set the property server.max-http-request-header-size to 24KB. Previous default value was 8KB.

Reviewer 3: As I understand this new property contains the default value. Is it necessary to give the hint that this value is the default value and can be changed if necessary!?

Author: Previously it was not set by us, but we were using the default value. Now we set it explicitly. I'll remove "default" to avoid confusion.
